.circleci/config.yml

```diff
@@ -0,0 +1,117 @@
+version: 2.1
+orbs:
+  node: circleci/node@4.1.0
+
+references:
+  persist-workspace: &persist-workspace
+    persist_to_workspace:
+      root: ~/source
+      paths:
+        - .
+
+  attach-workspace: &attach-workspace
+    attach_workspace:
+      at: .
+
+defaults: &defaults
+  working_directory: ~/source
+  docker:
+    - image: circleci/node:latest
+  resource_class: large
+
+jobs:
+  lint-test:
+    <<: *defaults
+    steps:
+      - checkout
+      - restore_cache:
+          keys:
+            - v1-dependencies-{{ checksum "package.json" }}
+            # fallback to using the latest cache if no exact match is found
+            - v1-dependencies-
+
+      - run:
+          name: Install dependencies
+          command: yarn install --frozen-lockfile
+      - run:
+          name: Run linting
+          command: yarn lint
+      - run:
+          name: Run tests
+          command: yarn test
+
+      - save_cache:
+          paths:
+            - node_modules
+          key: v1-dependencies-{{ checksum "package.json" }}
+      - *persist-workspace
+  build:
+    <<: *defaults
+    steps:
+      - *attach-workspace
+      - restore_cache:
+          keys:
+            - v1-dependencies-{{ checksum "package.json" }}
+            # fallback to using the latest cache if no exact match is found
+            - v1-dependencies-
+
+      - run:
+          name: Run build
+          command: yarn build
+          no_output_timeout: 20m
+
+      - *persist-workspace
+  release-publish:
+    <<: *defaults
+    steps:
+      - add_ssh_keys
+      - *attach-workspace
+      - restore_cache:
+          keys:
+            - v1-dependencies-{{ checksum "package.json" }}
+            # fallback to using the latest cache if no exact match is found
+            - v1-dependencies-
+
+      - run:
+          name: Setup .env
+          command: |
+            echo "GH_TOKEN=${CIRCLECI_MOONSTONE_RELEASE_TOKEN}" > ~/source/.env
+            echo "NPM_TOKEN=${NPMJS_TOKEN}" >> ~/source/.env
+      - run:
+          name: Authenticate with registry
+          command: echo "//registry.npmjs.org/:_authToken=$NPMJS_TOKEN" > ~/source/.npmrc
+      - run:
+          name: Release and publish
+          command: |
+            mkdir ~/.ssh
+            ssh-keyscan github.com >> ~/.ssh/known_hosts
+            git config user.email "jahia-ci@jahia.com"
+            git config user.name "Jahia CI"
+            git fetch --tags
+            yarn auto-changelog
+            yarn auto-version
+            npm publish --access public
+            git push --follow-tags --set-upstream origin master
+            yarn auto-release-only
+      - run:
+          name: Merge changes back into develop branch
+          command: |
+            git checkout develop
```

Comment:
where is the develop branch ?

Reply:
> It seems like a copy of moonstone config .. ?? Won't work here for sure. A more valid config was done here : #153
> https://jahia.slack.com/archives/G010WQ4N1B9

Yes I just copied it over so it will allow some merging
But definitely I can improve it and fix-it up to be better

Reply:
I will fix this portion-up

Reply:
Removed this portion as it is just copied